File: 2406-decode-the-message/2406-decode-the-message.cpp
class Solution {
public:
    string decodeMessage(string key, string message) {
        unordered_map<char, char> m;
        string ans = "";
        int i = 0;
        for(auto k : key){
            if(k != ' '){
                if(!m[k])
                    m[k] = 'a' + i++;
            }
            if(m.size()==26)break;
        }
        for(auto msg : message){
            if(msg == ' ')
                ans += " ";
            else
                ans += m[msg];
        }
        return ans;
    }
};

File: 2406-decode-the-message/2406-decode-the-message.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        d={}
        s="abcdefghijklmnopqrstuvwxyz"
        res={}
        k=97
        for i in s:
            d[i]=0
        for i in key:
            if(i==' '):
                continue
            if d[i]<=0:
                d[i]+=1
                # print(chr(k))
                res[i]=chr(k)
                k+=1
        result=''
        print(res)
        for i in message:
            if(i==' '):
                result+=' '
            else:
                if i in res:
                    result+=res[i]
        return result

